packages/router-api/src/primitives.rs

```diff
@@ -114,12 +114,33 @@ impl TryFrom<String> for Address {
     }
 }
 
+impl Address {
+    pub const fn is_address(value: &str) -> bool {
+        !value.is_empty() && !contains!(value, FIELD_DELIMITER)
```

Comment:
```suggestion
        nonempty::String::is_nonempty(value) && !contains!(value, FIELD_DELIMITER)
```
if we want to follow this pattern recursively to encapsulate the logic

Reply:
Could also start defining `*::is_valid` so usage becomes `Address::is_valid` instead `Address::is_address` although it doesn't feel as good since the argument itself isn't an Address but a string

Reply:
this check only affects the static compiler error, the actual parsing is still defined recursively. So I think I prefer to keep it simple, but I'll think about it